ambry-commons/src/test/java/com.github.ambry.commons/ByteBufferReadableStreamChannelTest.java

```diff
@@ -177,10 +180,75 @@ public void idempotentOperationsTest()
     assertFalse("ByteBufferReadableStreamChannel is not closed", byteBufferReadableStreamChannel.isOpen());
   }
 
+  /**
+   * Tests construction of {@link ByteBufferReadableStreamChannel} on some corner cases.
+   * <p/>
+   * Corner case list:
+   * 1. Create with a null {@link ByteBuffer}.
+   * 2. Create with a null list of {@link ByteBuffer}.
+   * 3. Create with an empty {@link ByteBuffer} list.
+   * 4. Create with a list of {@link ByteBuffer} containing a null {@link ByteBuffer}.
+   */
+  @Test
+  public void cornerCasesTest() {
```

Comment:
better called `constructionErrorCasesTest()` since these are not "corner" cases.